File: nexus-h1/generate_qr.py
import qrcode

with open('secrets/whatsapp_qr.txt', 'r') as f:
    qr_text = f.read().strip()

qr = qrcode.QRCode(version=1, box_size=10, border=2)
qr.add_data(qr_text)
qr.make(fit=True)

img = qr.make_image(fill_color="black", back_color="white")
img.save('secrets/whatsapp_qr.png')
print("QR code saved to secrets/whatsapp_qr.png")

File: nexus-h1/generate_qr.js
const QRCode = require('qrcode');
const fs = require('fs');

const qrText = fs.readFileSync('./secrets/whatsapp_qr.txt', 'utf8').trim();

QRCode.toFile('./secrets/whatsapp_qr.png', qrText, {
    width: 400,
    margin: 2
}, (err) => {
    if (err) {
        console.error('Error generating QR:', err);
        process.exit(1);
    }
    console.log('QR code saved to secrets/whatsapp_qr.png');
});